apps/services/utils.py

```diff
@@ -0,0 +1,76 @@
+from apps.common.exceptions import BackendException
+from apps.users.services import student_list_service
+
+from collections import defaultdict
+from fastapi import status as http_status
+import xlrd
+
+
+def check_content_type(file):
+    if file.content_type != "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet":
+        raise BackendException(
+            message="Uploaded file have invalid type.",
+            code=http_status.HTTP_406_NOT_ACCEPTABLE
+        )
+
+
+def create_faculty_dict(specialties):
+    faculty_dict = defaultdict(dict)
+    for specialty in specialties:
+        faculty_dict[specialty.shortname]["faculty_id"] = specialty.faculty_id
+        faculty_dict[specialty.shortname][specialty.name_1] = specialty.speciality_id
```

Comment:
what is name_1?

Reply:
When we creating query = select(Faculty, Speciality).filter(
        Speciality.faculty_id == Faculty.faculty_id
    ).where(Faculty.university_id == university_id),
second field with the same name automatically changed to <some_name>_1 (and so on).

Reply:
got it, thanks